apps/internal/oauth/ops/authority/authority.go

```diff
@@ -351,36 +352,52 @@ type Info struct {
 	InstanceDiscoveryDisabled bool
 }
 
-func firstPathSegment(u *url.URL) (string, error) {
-	pathParts := strings.Split(u.EscapedPath(), "/")
-	if len(pathParts) >= 2 {
-		return pathParts[1], nil
-	}
-
-	return "", errors.New(`authority must be an https URL such as "https://login.microsoftonline.com/<your tenant>"`)
-}
-
 // NewInfoFromAuthorityURI creates an AuthorityInfo instance from the authority URL provided.
 func NewInfoFromAuthorityURI(authority string, validateAuthority bool, instanceDiscoveryDisabled bool) (Info, error) {
 	u, err := url.Parse(strings.ToLower(authority))
 	if err != nil || u.Scheme != "https" {
 		return Info{}, errors.New(`authority must be an https URL such as "https://login.microsoftonline.com/<your tenant>"`)
 	}
 
-	tenant, err := firstPathSegment(u)
-	if err != nil {
-		return Info{}, err
+	// the old firstPathSegment didn't account for any other case than https://<url>/<tenant>
+	// which is insufficient to support dSTS flow
+	//
+	// I am not sure if we should go with the way of parsing the path and then checking
+	// its properties, or we could go for some regex matching?
```

Comment:
Sounds like premature optimization; I would go with standard library.

Reply:
I have seen another PR / issue in the msal repo that touched this exact part of code, trying to bend it for yet another use-case, but I can't seem to be able to find it now.. in short, I can imagine there being far more use-cases than just this one. I would definitely not call 'trying to find a more flexible approach' premature, otherwise we might end up with hard-coding some exotic use-case every time it pops up

btw, wdym by _standard library_? 😅 https://pkg.go.dev/regexp is this not std enough? 🙈